src/test/scala/com/databricks/spark/csv/util/TypeCastSuite.scala

```diff
@@ -115,4 +117,48 @@ class TypeCastSuite extends FunSuite {
     assert(TypeCast.castTo("", StringType, true, false, "") == "")
     assert(TypeCast.castTo("", StringType, true, true, "") == null)
   }
+
+  test("Parse exception is caught correctly"){
+
+    def testParseException( castType : DataType, badValues : Seq[String]): Unit = {
+      badValues.foreach(testValue => {
+        assert(TypeCast.castTo(testValue, castType, true, false, "", null, true) == null)
+        // if not nullable it isn't null
+        assert(Try(TypeCast.castTo(testValue, castType, false, false, "", null, true)).isFailure)
+      }
+      )
+    }
```

Comment:
Indentation and remove extra function here, (see, [scala-style-guide#anonymous-methods](https://github.com/databricks/scala-style-guide#anonymous-methods)) maybe as below:

```scala
def testParseException(castType : DataType, badValues : Seq[String]): Unit = {
  badValues.foreach { testValue =>
    assert(TypeCast.castTo(testValue, castType, true, false, "", null, true) == null)
    // if not nullable it isn't null
    assert(Try(TypeCast.castTo(testValue, castType, false, false, "", null, true)).isFailure)
  }
}
```